func_sig_registry/registry/views.py

```diff
@@ -77,6 +81,41 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class EventSignatureListView(SingleTableView, ListView):
+    model = EventSignature
+    table_class = EventSignatureTable
+    table_pagination = {
+        'per_page': 10
+    }
+
+    def get_queryset(self):
+        queryset = super(EventSignatureListView, self).get_queryset()
+
+        if self.request.GET.get('bytes_signature'):
+            hex_signature = self.request.GET['bytes_signature']
+            unprefixed_hex_signature = remove_0x_prefix(hex_signature.lower())
+
+            if len(unprefixed_hex_signature) == 64:
+                return queryset.filter(
+                    hex_signature=unprefixed_hex_signature,
+                )
+            else:
+                return queryset.filter(
+                    hex_signature__icontains=unprefixed_hex_signature,  #NOQA
+                )
+        return queryset
+
+    def get_context_data(self, **kwargs):
+        context = super(EventSignatureListView, self).get_context_data(**kwargs)
+        if self.request.GET.get('hex_signature'):
+            serializer = EventSignatureSearchForm(data=self.request.GET)
+            serializer.is_valid()
+        else:
+            serializer = EventSignatureSearchForm()
+        context['serializer'] = EventSignatureSearchForm()
```

Comment:
This looks like an error.  You are probably intending something like `context['serializer'] = serializer`

Before fixing this, I would recommend looking into writing a test that would have exposed this bug.  I don't think we currently have any tests that interact with these views so there isn't any boilerplate for you to start from.  I'd recommend looking at the test client that django provides.

https://docs.djangoproject.com/en/3.1/topics/testing/tools/#the-test-client

Reply:
Good point! I will look into this and add tests for functions as well.

Reply:
I tried to run different tests, so far it looks like this code puts the requested text signature back to search input and renders it with the search results. It is all, or am I missing something?

Reply:
Notice in the code above that you initialize `serializer` in the `if/else` blocks but then that variable is not used.  Instead a new instance of the serializer (without the request data is instantiated and placed into `context`.